codec/optional_packer.go

```diff
@@ -9,21 +9,31 @@ import (
 	"github.com/ava-labs/hypersdk/crypto"
 )
 
+// Represents a bitmask.
 // Inspired by: https://yourbasic.org/golang/bitmask-flag-set-clear
 type bits uint8
```

Comment:
We should probably just replace this with `BitSet`: https://github.com/ava-labs/avalanchego/blob/c5d8cd3b17747a6dfb3eae71de711a764ebd69f5/utils/set/bits.go#L12-L102

Would be great if we did that migration here.

Reply:
We can check that an item is properly represented by `b.Bytes() == bytes`

Reply:
We can also check that the BitLen is never greater than the registered bits.

Reply:
updated